libs/watt/src/lib/components/field/watt-field.component.ts

```diff
@@ -14,88 +14,96 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
-  viewProviders: [{ provide: ControlContainer, useExisting: FormGroupDirective }],
+  imports: [WattIconComponent, WattTooltipDirective, WattFieldErrorComponent, VaterStackComponent],
   styleUrls: ['./watt-field.component.scss'],
   template: `
-    <label [attr.for]="id ? id : null">
-      @if (!chipMode && label) {
-        <span class="label" [ngClass]="{ required: _isRequired }">
-          {{ label }}
-          @if (tooltip) {
+    <label [attr.for]="id()">
+      @if (!chipMode() && label()) {
+        <span class="label" [class.required]="isRequired()">
+          {{ label() }}
+          @if (tooltip(); as tooltip) {
             <watt-icon name="info" wattTooltipPosition="top" [wattTooltip]="tooltip" />
           }
         </span>
       }
-      <div style="display: flex;align-items: center; gap: var(--watt-space-s);">
+      <vater-stack direction="row" gap="s">
         <div class="watt-field-wrapper" #wrapper>
           <ng-content />
         </div>
         <ng-content select="watt-field-descriptor" />
-      </div>
+      </vater-stack>
       <ng-content select="watt-field-hint" />
       <ng-content select="watt-field-error" />
-      @if (control?.errors?.['required'] || control?.errors?.['rangeRequired']) {
+      @if (isEmpty()) {
         <watt-field-error>{{ intl.required }}</watt-field-error>
       }
     </label>
   `,
+  host: {
+    '[class.watt-field--chip]': 'chipMode()',
+  },
 })
-export class WattFieldComponent implements OnChanges {
-  private _formGroupDirective = inject(FormGroupDirective, { optional: true });
+export class WattFieldComponent {
   intl = inject(WattFieldIntlService);
 
-  @Input() label: string | undefined;
-  @Input({ required: true }) control!: FormControl | null;
-  @Input() id!: string;
-  @Input() chipMode = false;
-  @Input() tooltip?: string;
+  control = input<FormControl | null>(null);
+  label = input<string>();
+  id = input<string>();
+  chipMode = input(false);
+  tooltip = input<string>();
 
-  // Used for text fields with autocomplete
-  @ViewChild('wrapper', { static: true }) wrapper!: ElementRef<HTMLDivElement>;
+  errors = signal<ValidationErrors | null>(null);
+  isRequired = signal(false);
+  isEmpty = computed(() => this.errors()?.['required'] || this.errors()?.['rangeRequired']);
 
-  @HostBinding('class.watt-field--chip')
-  get _chip() {
-    return this.chipMode;
-  }
+  // Used for text fields with autocomplete
+  wrapper = viewChild.required<ElementRef>('wrapper');
 
-  @HostBinding('class.watt-field--invalid')
-  get _hasError() {
-    return (
-      (this.control?.status === 'INVALID' && !!this.control?.touched) ||
-      (this._formGroupDirective?.submitted && this.control?.status === 'INVALID')
+  constructor() {
+    const status$ = toObservable(this.control).pipe(
+      filter((control) => control !== null),
```

Comment:
Question: Should there be a unsubscribe somewhere? Or a takeUntilDestoryed?

Reply:
No need, it completes itself:
https://github.com/angular/angular/blob/main/packages/core/rxjs-interop/src/to_observable.ts#L65

Reply:
Perfect!